### PR TITLE
feat: add fallback proposal mechanism to L2OutputOracle

### PR DESCRIPTION
**Pull request by puma314**
## Summary
- Add fallback proposal mechanism that allows anyone to submit proposals if a configurable timeout period has passed since the last proposal
- Add comprehensive unit tests covering all scenarios of the fallback functionality

### Core Implementation
- Added `fallbackProposalTimeout` parameter to contract initialization
- Added `lastProposalTimestamp()` helper function to get timestamp of last proposal  
- Modified both `proposeL2Output` functions to bypass approved proposer check when timeout exceeded
- Added getter function `FALLBACK_PROPOSAL_TIMEOUT()` following existing patterns

### Test Coverage
- Updated test infrastructure to support `fallbackProposalTimeout` parameter
- Added comprehensive test suite `OPSuccinctL2OutputOracleFallbackTest` with 7 test cases:
  - ✅ Non-approved proposer can propose when timeout elapsed
  - ✅ Non-approved proposer cannot propose when timeout not elapsed
  - ✅ Approved proposers can still propose before timeout
  - ✅ Approved proposers can still propose after timeout  
  - ✅ `lastProposalTimestamp()` returns correct values
  - ✅ Getter functions work correctly

## Benefits
- **Liveness guarantee**: Prevents protocol stalls when approved proposers are inactive
- **Configurable**: Timeout period can be set during deployment (suggested: 2 days)
- **Backward compatible**: Doesn't affect normal operation with approved proposers
- **Well tested**: Comprehensive test coverage ensures correct behavior

## Test Plan
- [x] Unit tests pass for all fallback scenarios
- [x] Normal proposer functionality remains unchanged
- [x] Edge cases around timeout boundary are covered
- [x] Helper functions work correctly

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: contracts/src/validity/OPSuccinctL2OutputOracle.sol [hunk 1]
```diff
@@ -274,6 +280,14 @@ contract OPSuccinctL2OutputOracle is Initializable, ISemver {
...
     }
 
+    /// @notice Getter for the fallbackProposalTimeout.
+    ///         Public getter is legacy and will be removed in the future. Use `fallbackProposalTimeout` instead.
+    /// @return Fallback proposal timeout in seconds.
+    /// @custom:legacy
+    function FALLBACK_PROPOSAL_TIMEOUT() external view returns (uint256) {
+        return fallbackProposalTimeout;
+    }
+
```
Comment (fakedev9999) [inline contracts/src/validity/OPSuccinctL2OutputOracle.sol]: nit: I think we don't need this.

**Review by fakedev9999 — APPROVED** [review]
LGTM

File: contracts/src/fp/AccessManager.sol [hunk 2]
```diff
@@ -26,6 +26,28 @@ contract AccessManager is Ownable {
     /// @notice Tracks whitelisted challengers.
     mapping(address => bool) public challengers;
 
+    /// @notice The timeout (in seconds) after which permissionless proposing is allowed (immutable).
+    uint256 public immutable permissionlessTimeout;
```
Comment (fakedev9999) [inline contracts/src/fp/AccessManager.sol]: Why is this immutable while `fallbackProposalTimeout` in the L2OO is not immutable?

**Review by fakedev9999 — CHANGES_REQUESTED** [review]

**Review by fakedev9999 — APPROVED** [review]
LGTM! Let's rebase and merge :)
